### PR TITLE
Fix problems with Constructor Calls and make Constructor tests more thorough

### PR DESCRIPTION
**Pull request by VoR0220**
* fixes https://github.com/eris-ltd/eris-cli/issues/1065
* gets deploy jobs using the same code as call and query jobs for preprocessing (thus making it look cleaner)
* gets ABI to handle the case of a constructor in core.go
* fleshes out constructor tests to test for multiple types and different methods of input in a epm.yaml file.

Signed-off-by: VoR0220 <rj@erisindustries.com>

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ machine:
     REPO: ${GOPATH}/src/github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}
     DOCKER_VERSION: 1.9.1
     DOCKER_MACHINE_VERSION: 0.7.0
-    ERIS_CLI_BRANCH: develop
+    ERIS_CLI_BRANCH: master
```
Comment (zramsay) [inline circle.yml]: why?

Comment (VoR0220) [inline circle.yml]: it wouldn't pass unless it was on the master branch.

Comment (VoR0220) [inline circle.yml]: And therefore would not deploy.

**Comment by zramsay** [comment]
LGTM. blocking on documentation noted above. otherwise :+1:

**Pull request merged**